director.go

```diff
@@ -129,16 +136,38 @@ func QueryDirector(source string, directorUrl string) (resp *http.Response, err
 		},
 	}
 
-	log.Debugln("Querying OSDF Director at", resourceUrl)
-	resp, err = client.Get(resourceUrl)
-	log.Debugln("Director's response:", resp)
+	req, err := http.NewRequest("GET", resourceUrl, nil)
+	if err != nil {
+		log.Errorln("Failed to create an HTTP request:", err)
+		return nil, err
+	}
+
+	// Include the Client's version as a User-Agent header. The Director will decide
+	// if it supports the version, and provide an error message in the case that it
+	// cannot.
+	userAgent := "pelican-client/" + ObjectClientOptions.Version
+	req.Header.Set("User-Agent", userAgent)
 
+	// Perform the HTTP request
+	resp, err = client.Do(req)
 	if err != nil {
-		log.Errorln("Failed to get response from OSDF Director:", err)
+		log.Errorln("Failed to get response from the director:", err)
 		return
 	}
-
 	defer resp.Body.Close()
+	log.Debugln("Director's response:", resp)
+
+	// Check HTTP response -- should be 307 (redirect), else something went wrong
+	body, _ := io.ReadAll(resp.Body)
+	if resp.StatusCode != 307 {
+		fmt.Printf("\n\n\nresp.StatusCode %d\n\n\n", resp.StatusCode)
```

Comment:
Why so many newlines? Just for formatting?

Reply:
An errant debug comment! Thanks for catching this!